src/BenchmarkDotNet.Core/Toolchains/InProcess/InProcessExecutor.cs

```diff
@@ -0,0 +1,150 @@
+
+namespace BenchmarkDotNet.Toolchains.InProcess
+{
+    /// <summary>
+    ///     Implementation of <see cref="IExecutor" /> for in-process benchmarks.
+    /// </summary>
+    [PublicAPI]
+    [SuppressMessage("ReSharper", "ArrangeBraces_using")]
+    public class InProcessExecutor : IExecutor
+    {
+        private static readonly TimeSpan UnderDebuggerTimeout = TimeSpan.FromDays(1);
+
+        /// <summary> Default timeout for in-process benchmarks. </summary>
+        public static readonly TimeSpan DefaultTimeout = TimeSpan.FromMinutes(5);
+
+        #region .ctor & properties
+
+        /// <summary>Initializes a new instance of the <see cref="InProcessExecutor" /> class.</summary>
+        /// <param name="timeout">Timeout for the run.</param>
+        /// <param name="codegenMode">Describes how benchmark action code is generated.</param>
+        /// <param name="logOutput"><c>true</c> if the output should be logged.</param>
+        public InProcessExecutor(TimeSpan timeout, BenchmarkActionCodegen codegenMode, bool logOutput)
+        {
+            if (timeout == TimeSpan.Zero)
+                timeout = DefaultTimeout;
+
+            ExecutionTimeout = timeout;
+            CodegenMode = codegenMode;
+            LogOutput = logOutput;
+        }
+
+        /// <summary>Timeout for the run.</summary>
+        /// <value>The timeout for the run.</value>
+        public TimeSpan ExecutionTimeout { get; }
+
+        /// <summary>Describes how benchmark action code is generated.</summary>
+        /// <value>Benchmark action code generation mode.</value>
+        public BenchmarkActionCodegen CodegenMode { get; }
+
+        /// <summary>Gets a value indicating whether the output should be logged.</summary>
+        /// <value><c>true</c> if the output should be logged; otherwise, <c>false</c>.</value>
+        public bool LogOutput { get; }
+
+        #endregion
+
+        /// <summary>Executes the specified benchmark.</summary>
+        /// <param name="buildResult">The build result.</param>
+        /// <param name="benchmark">The benchmark.</param>
+        /// <param name="logger">The logger.</param>
+        /// <param name="resolver">The resolver.</param>
+        /// <param name="diagnoser">The diagnoser.</param>
+        /// <returns>Execution result.</returns>
+        public ExecuteResult Execute(
+            BuildResult buildResult, Benchmark benchmark, ILogger logger, IResolver resolver, IDiagnoser diagnoser = null)
+        {
+            // TODO: preallocate buffer for output (no direct logging)?
+            var host = new InProcessHost(benchmark, LogOutput ? logger : null, diagnoser);
+
+            int exitCode = -1;
+            var runThread = new Thread(() => exitCode = ExecuteCore(host, benchmark, logger));
+            if (benchmark.Target.Method.GetCustomAttributes<STAThreadAttribute>(false).Any())
+            {
+                // TODO: runThread.SetApartmentState(ApartmentState.STA);
```

Comment:
it's great that you remember that we support STA, may I ask why is this commented out?